java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -131,6 +170,58 @@ public Block(int id) {
     public Tree terminator() {
       return terminator;
     }
+
+    public boolean notActive() {
+      return terminator == null && elements.isEmpty();
+    }
+
+    private void prune(Block inactiveBlock) {
+      if (inactiveBlock == trueBlock) {
+        if (inactiveBlock.successors.size() != 1) {
+          throw new IllegalStateException("True successor must be replaced by a uniuqe successor!");
+        }
+        trueBlock = inactiveBlock.successors.iterator().next();
+      }
+      if (inactiveBlock == falseBlock) {
+        if (inactiveBlock.successors.size() != 1) {
+          throw new IllegalStateException("True successor must be replaced by a uniuqe successor!");
+        }
+        falseBlock = inactiveBlock.successors.iterator().next();
+      }
+      if (successors.remove(inactiveBlock)) {
+        successors.addAll(inactiveBlock.successors);
+      }
+    }
+
+    @Override
+    public String toString() {
+      StringOutputStream buffer = new StringOutputStream();
+      debugTo(new PrintStream(buffer));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this use of constructor "PrintStream(OutputStream)" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1943)